File: content/courses/parallel-computing-introduction/codes/omp_truedep.c
#include <stdio.h>
#include <omp.h>

int main() {

    const int N=48;
    int a[N];

    a[0]=100;

#pragma omp parallel for 
    for (int i=1; i<N; i++) {
        a[i] = a[i-1] + 10;
    }

    for (int i=0; i<N; i++) {
        printf("%d %d\n",i,a[i]);
    }

}

File: content/courses/parallel-computing-introduction/codes/omp_truedep.f90
program truedep

    integer, parameter    :: N=48
    integer, dimension(N) :: a
    integer               :: x=5
    integer               :: i

    a(1)=100

    !$omp parallel do
    do i=1, N-1
        a(i+1)=a(i)+10
    enddo
    !$omp end parallel do

    do i=1, N
       write(*,*) i,a(i)
    enddo

end program
